apps/communications/dialer/test/marionette/lib/dialer.js

```diff
@@ -1,78 +1,74 @@
 'use strict';
 
-/**
- * Abstraction around dialer app.
- * @constructor
- * @param {Marionette.Client} client for operations.
- */
-function Dialer(client) {
-  this.client = client;
-}
-
-/**
- * @type String Origin of dialer app
- */
-Dialer.URL = 'app://communications.gaiamobile.org';
-
-Dialer.config = {
-  settings: {
-    'devtools.overlay': true,
-    'hud.reflows': true
-  },
-  prefs: {
-    'devtools.debugger.forbid-certified-apps': false
-  }
-};
-
-Dialer.Selectors = {
-  phoneNumber: '#phone-number-view',
-
-  keypadView: '#keyboard-view',
-  del: '#keypad-delete',
-  zero: '.keypad-key[data-value="0"]',
-  one: '.keypad-key[data-value="1"]',
-  two: '.keypad-key[data-value="2"]',
-  three: '.keypad-key[data-value="3"]',
-  keypadCallBarAddContact: '#keypad-callbar-add-contact',
-
-  callLogTabItem: '#option-recents',
-  callLogEditButton: '#call-log-icon-edit',
-  callLogTabs: '#call-log-filter',
-  callLogNoResultsContainer: '#no-result-container',
-  callLogItem: '.log-item',
-  callLogEditForm: '#edit-mode',
-
-  contactsTabItem: '#option-contacts',
-  contactsIframe: '#iframe-contacts',
-
-  addToExistingContactMenuItem: 'button[data-l10n-id="addToExistingContact"]'
-};
-
-Dialer.prototype = {
-  /**
-   * Launches dialer app and focuses on frame.
-   */
-  launch: function() {
-    this.client.apps.launch(Dialer.URL, 'dialer');
-    this.switchTo();
-    this.client.helper.waitForElement('body');
-  },
-
-  relaunch: function() {
-    this.client.apps.close(Dialer.URL, 'dialer');
-    this.launch();
-  },
-
-  switchTo: function() {
-    this.client.switchToFrame();
-    // switchToApp already waits for the app to be displayed
-    this.client.apps.switchToApp(Dialer.URL, 'dialer');
-  },
-
-  get phoneNumber() {
-    return this.client.helper.waitForElement(Dialer.Selectors.phoneNumber)
-                             .getAttribute('value');
-  }
-};
-
-module.exports = Dialer;
+var CallLogAccessor = require('./views/call_log/accessors');
+var DialerOptionsAccessor = require('./views/dialer_options/accessors');
+var KeypadAccessors = require('./views/keypad/accessors');
+
+(function(module) {
+
+  var ORIGIN_URL = 'app://communications.gaiamobile.org';
+  var MANIFEST_URL = ORIGIN_URL + '/manifest.webapp';
+
+  var SELECTORS = Object.freeze({
+    main: 'body'
+    //TBD
+  });
+
+  module.exports = {
+    create: function(client) {
+      var actions = client.loader.getActions();
+      //var newMessageView = new NewMessageView(client);
+
+      return {
+        Selectors: SELECTORS,
+
+        manifestURL: MANIFEST_URL,
+
+        DialerOptions: new DialerOptionsAccessor(client),
```

Comment:
Yes, I will keep it as it was originally and try to not to modify it to much. 
Btw, I should have told you, sorry, the save_contact.js does not follow the class pattern, was just to check that I was able to write it and it worked before trying to implement the PC pattern. The test using the new pattern is the add_contact_PO.js